### PR TITLE
Move most packages into internal

### PR DESCRIPTION
**Pull request by ameyer-pivotal**
Moves most packages into internal, as a first step toward reorganizing our packages.

Phase 2, which will be a later PR, will be to minimize/simplify the packages within `internal`.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -37,21 +38,21 @@ func WithLogger(l logging.Logger) ClientOption {
...
-func WithFetcher(f ImageFetcher) ClientOption {
+func WithFetcher(f internal.ImageFetcher) ClientOption {
 	return func(c *Client) {
 		c.imageFetcher = f
 	}
 }
 
 // WithDownloader supply your own downloader.
-func WithDownloader(d Downloader) ClientOption {
+func WithDownloader(d internal.Downloader) ClientOption {
```
Comment (jromero) [inline client.go]: A general rule should be that anything _used_ by a public API cannot reference an internal component. This is an example of where that is violated. Another example would be returning an internal object/interface from a public method.

Comment (ameyer-pivotal) [reply client.go]: I agree, and missed this. I also missed that the package here ended up as `internal`, which I believe is not recommended.

Comment (ameyer-pivotal) [reply client.go]: Fixed.

File: create_builder.go [hunk 2]
```diff
@@ -8,15 +8,16 @@ import (
...
+	"github.com/buildpack/pack/internal/builder"
+	"github.com/buildpack/pack/internal/dist"
+	"github.com/buildpack/pack/internal/image"
+	"github.com/buildpack/pack/internal/style"
 )
 
 type CreateBuilderOptions struct {
 	BuilderName   string
-	BuilderConfig builder.Config
+	BuilderConfig builder2.Config
```
Comment (jromero) [inline create_builder.go]: :| Not a fan of `builder2`. Feel like if we are going to be running into this issue more frequently we should have a standard way of differentiating them.

Comment (ameyer-pivotal) [reply create_builder.go]: Agreed. That's definitely not intentional.

File: internal/buildpackage/buildpackage.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package buildpackage
+
+import (
+	dist2 "github.com/buildpack/pack/internal/dist"
```
Comment (jromero) [inline internal/buildpackage/buildpackage.go]: Why is this `dist2`?

File: internal/commands/inspect_builder.go [hunk 4]
```diff
@@ -12,12 +12,12 @@ import (
...
 
 	"github.com/buildpack/pack"
-	"github.com/buildpack/pack/api"
-	"github.com/buildpack/pack/builder"
-	"github.com/buildpack/pack/config"
-	"github.com/buildpack/pack/dist"
+	"github.com/buildpack/pack/internal/api"
+	"github.com/buildpack/pack/internal/builder"
+	"github.com/buildpack/pack/internal/config"
+	dist2 "github.com/buildpack/pack/internal/dist"
```
Comment (jromero) [inline internal/commands/inspect_builder.go]: `dist2`?

File: interfaces.go [hunk 5]
```diff
@@ -1,30 +1,7 @@
 package pack
```
Comment (jromero) [inline interfaces.go]: The remnants here look out of place.

File: client.go [hunk 6]
```diff
@@ -1,20 +1,37 @@
...
-	"github.com/buildpack/pack/config"
-	"github.com/buildpack/pack/image"
+	"github.com/buildpack/pack/internal/blob"
+	"github.com/buildpack/pack/internal/build"
+	"github.com/buildpack/pack/internal/config"
+	"github.com/buildpack/pack/internal/image"
 	"github.com/buildpack/pack/logging"
 )
 
+type ImageFetcher interface {
```
Comment (jromero) [inline client.go]: These are missing their mock generation comments.

File: internal/buildpackage/buildpackage.go (hunk 3 above)
Comment (ameyer-pivotal) [reply internal/buildpackage/buildpackage.go]: Something automatic by the IDE. Will fix. Good catch!

File: internal/commands/inspect_builder.go (hunk 4 above)
Comment (ameyer-pivotal) [reply internal/commands/inspect_builder.go]: Same as other comment, will fix.

File: client.go (hunk 6 above)
Comment (ameyer-pivotal) [reply client.go]: They're in `interfaces.go`. Happy to fix, though this was an issue before this PR

File: interfaces.go (hunk 5 above)
Comment (ameyer-pivotal) [reply interfaces.go]: See response in `client.go`.

Comment (ameyer-pivotal) [reply interfaces.go]: Agreed it's sort of weird to have them here, and I'm happy to disperse them to their homes.

Comment (ameyer-pivotal) [reply interfaces.go]: I think these were grouped here for convenience, but now that we have `make generate`, there's no need.

**Pull request merged**